app/models/tag.rb

```diff
@@ -39,17 +39,11 @@ def cognates_list
   #
   # @param str_list [Array<String>] list of tag names to set as cognates
   def cognates_list=(str_list)
-    if persisted?
-      remove_cognates(str_list)
-      create_cognates(str_list)
-    else
-      self.tag_cognates_attributes = str_list.filter_map do |name|
-        next if name.blank?
-
-        cognate = Tag.find_or_create_with_like_by_name(name)
-        { cognate_id: cognate.id } if cognate.id != id
-      end
-    end
+    names = str_list.compact_blank.uniq.reject { |name| name == self.name }
```

Comment:
If we use other _name_ for block parameter, attribute _name_ will not be shadowed